libraries/chain/backing_store/db_context_rocksdb.cpp

```diff
@@ -252,6 +254,7 @@ namespace eosio { namespace chain { namespace backing_store {
          table_store.table, old_payer, payer, name(key_store.primary),
          old_key_value.value->data(), old_key_value.value->size(), value, value_size);
       }
+      primary_iter_store.set_value(itr, value, value_size);
```

Comment:
Should this be primary_iter_store.set_value(iterator, pp.value, pp.value_size); ?

Reply:
no, itr is the "handle" for the identified code/scope/table/primary_key, and update cannot change any of those parameters, to do that you would need to call db_rem_i64 code/scope/table/primary_key then call db_store_i64 code/scope/table/primary_key2

Reply:
It could be, value and pp.value and value_size and pp.value_size are just copies of each other. The payer_payload makes a copy and then when you call pp.to_payload() it returns a packed buffer with both the payer and value data in it. But since we have value and value_size passed in, we can just use it.